### PR TITLE
[FIX] Solving the navigation issue from threads on tablets

### LINKED ISSUE #3969: [bug] Thread navigation issue on tablets
### Description:

Probably that bug is only tablet issue.
When user opens some thread in chat and then navigate to another chat in sidebar, he can't return to first chat.

### Environment Information:

- Rocket.Chat Server Version: 4.3.3
- Rocket.Chat App Version: 4.25.0.29396
- Device Name: iPad Pro
- OS Version: 15.3.1

### Steps to reproduce:

1. Go to some thread in chatA on iPad in tablet view
2. Tap on any other chat in sidebar
3. Tap in chatA in sidebar
4. Nothing happens - chatA don't opens

### Expected behavior:

User should be able to open chatA after navigated another chat

### Actual behavior:

https://user-images.githubusercontent.com/4023037/160352181-c346707b-cfbf-41de-abdb-99b8855e09ae.MP4

### Additional context:

No

### PR DESCRIPTION
**Pull request by samay-kothari**
## Proposed changes

When navigating from a thread in a room to a different room directly using the side pane on tablet was not unsubscribing the current room. This fix makes sure that the user is not subscribed to more than one room at a single time. It removes the extra room from the room redux component.
## Issue(s)	

Closes #3969 
## Screenshots

https://user-images.githubusercontent.com/55805094/162345029-6961b8dd-26d5-4f25-ab15-e92140c2b010.mov

## Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] Improvement (non-breaking change which improves a current function)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation update (if none of the other choices apply)

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/RocketChat/Rocket.Chat/blob/develop/.github/CONTRIBUTING.md#contributing-to-rocketchat) doc
- [x] I have signed the [CLA](https://cla-assistant.io/RocketChat/Rocket.Chat.ReactNative)
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works (if applicable)
- [ ] I have added necessary documentation (if applicable)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by samay-kothari** [comment]
Hi @gerzonc,
Can you review this?

File: app/lib/methods/subscriptions/room.ts [hunk 1]
```diff
@@ -69,6 +69,12 @@ export default class RoomSubscription {
 		}
 
 		reduxStore.dispatch(subscribeRoom(this.rid));
+		const rooms = reduxStore.getState().room;
+		if (rooms.rooms.length > 1) {
+			for (let i = 1; i < rooms.rooms.length; i++) {
+				reduxStore.dispatch(unsubscribeRoom(rooms.rooms[i]));
+			}
+		}
```
Comment (gerzonc) [inline app/lib/methods/subscriptions/room.ts:77]: If this solves a problem only for tablets then it should be wrapped around an if validation that checks the current device platform (e.g. `if (isTablet)`). We already count with `isTablet`, `isAndroid` and `isiOS`

**Review by gerzonc — CHANGES_REQUESTED** [review]

**Review by gerzonc — APPROVED** [review]
LGTM

**Comment by ankar84** [comment]
@gerzonc @diegolmello @GleidsonDaniel hey!
Is there a plans to land that fix to next release?
Thanks!

**Comment by ankar84** [comment]
Any news?

**Comment by diegolmello** [comment]
@ankar84 This is not the _best_ way to fix it, so I need to try another way.
For example, disabling nested rooms (Room A opening Room B) would solve this by default.

**Comment by ankar84** [comment]
> @ankar84 This is not the _best_ way to fix it, so I need to try another way. For example, disabling nested rooms (Room A opening Room B) would solve this by default.

ok. I get it. Waiting for a proper fix then.
Thank you, Diego!

**Pull request closed without merging**